core/src/main/java/hudson/tools/InstallerTranslator.java

```diff
@@ -69,8 +73,16 @@ public String getToolHome(Node node, ToolInstallation tool, TaskListener log) th
                 } finally {
                     semaphore.release();
                 }
+            } else {
+                inapplicableInstallersMessages.add(Messages.CannotBeInstalled(
+                        installer.getDescriptor().getDisplayName(),
+                        tool.getName(),
+                        node.getDisplayName()));
             }
         }
+        for (String message : inapplicableInstallersMessages) {
+            log.getLogger().println(message);
```

Comment:
Ideally it should be a single message for a correct logging. Especially once we have the External task logging support. Should be fine for now